data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/targets/PublishedNewsDisplayedPropUpgrade.java

```diff
@@ -75,26 +76,29 @@ public int getPublishedNewsDisplayedPropCount() {
   public void processUpgrade(String oldVersion, String newVersion) {
     long startupTime = System.currentTimeMillis();
     LOG.info("Start upgrade of published news displayed property");
-
+    List<Metadata> newsTargetMetadatas = metadataService.getMetadatas(NewsTargetingService.METADATA_TYPE.getName(), 0);
+    List<MetadataItemEntity> newsTargetsMetadataItems = new ArrayList<>();
+    ExoContainerContext.setCurrentContainer(container);
+    for (Metadata newsTargetMetadata : newsTargetMetadatas) {
+      RequestLifeCycle.begin(entityManagerService);
+      EntityManager entityManager = entityManagerService.getEntityManager();
+      StringBuilder newsTargetsMetadataItemsQueryStringBuilder = new StringBuilder("SELECT * FROM SOC_METADATA_ITEMS WHERE METADATA_ID = '");
+      newsTargetsMetadataItemsQueryStringBuilder.append(newsTargetMetadata.getId() + "'");
```

Comment:
IMO it's better to add a single query with `METADATA_ID IN ('ID1', 'ID2', 'ID3' ...)`